nni/contrib/compression/quantization/lsq_quantizer.py

```diff
@@ -116,7 +116,16 @@ def register_scale(self):
             for target_name, _ in ts.items():
                 if hasattr(wrapper, f"{target_name}_scale"):
                     delattr(wrapper, f"{target_name}_scale")
-                param = torch.nn.Parameter()
+                # for deepspeed
+                try:
+                    device = next(wrapper.parameters()).device
+                except StopIteration:
+                    try:
+                        device = next(wrapper.buffers()).device
+                    except StopIteration:
+                        # NOTE: this will have risk in model parallel
+                        device = next(self.bound_model.parameters()).device
+                param = torch.nn.Parameter(torch.Tensor([1.0]).to(device))
```

Comment:
why deepspeed needs this?

Reply:
The other quantization algorithms do not have this issue. This problem is mainly caused by Deepspeed's need to allocate buffers for parameters. The previous implementation registered an empty parameter, which would lead to errors.